register/views.py

```diff
@@ -85,24 +101,153 @@ def user_verify(request, uidb64, token):
 
         return redirect('register:login')
 
-def user_login(request):
-    if request.method == 'POST':
-        username = request.POST.get('username')
-        password = request.POST.get('password')
-
-        user = authenticate(username=username, password=password)
 
-        
-
-        if user:
+def user_login(request):
+    if request.user.is_authenticated:
+        messages.success(request, 'Logged in successfully!')
+        return render(request, 'mainapp/home.html', {'message': 'Logged in successfully!'})
+    if request.method == "POST":
+        username=request.POST['username']
+        password=request.POST['password']
+        user=authenticate(username=username, password=password)
+        if user is not None:
             if user.is_active:
                 login(request, user)
-                return render(request, 'mainapp/home.html', {'message': 'Logged in successfully!'})
+                messages.success(request, 'Logged in successfully!')
+                return render(request,  'mainapp/home.html', {'message': 'Logged in successfully!'})
+            else:
+                messages.warning(request, 'You have not activated your account, activate using the link sent to your mail.')
+                return render(request, 'register/login.html', {'error': 'You have not activated your account, activate using the link sent to your mail.'})
         else:
-            messages.warning(request, 'You need to register yourself first!')
-            return redirect('/register/')
+            try:
+                user=User.objects.get(username=username)
+                if not user.is_active:
+                    messages.warning(request, 'You have not activated your account, activate using the link sent to your mail.')
+                    return render(request, 'register/login.html', {'error': 'You have not activated your account, activate using the link sent to your mail.'})
+                if password!=user.password:
```

Comment:
this is not how the password is compared. Use [this SO answer](https://stackoverflow.com/a/55822545/13398404) and use `check_password()` function to do it properly.